### DISCUSSION TITLE
How can I route all traffics through warp except few?

### DISCUSSION THREAD

**Discussion by maxwelldorliea** [XTLS/Xray-core#3386] [Q&A] [answered]
I recently started routing all of my traffic through Warp but then my clients started to face issues with their IPTV. When I stopped the routing of all traffic through Warp their IPTV worked and when I did the opposite their IPTV stopped working. I currently know the domain name and port of the IPTV services they (my clients) are using. How can I exclude traffic to these domains from routing through Warp?

This is the routing rule I am currently using to route all traffic through warps. Thanks to this answer @hassansim [Hassansim Answer](https://github.com/XTLS/Xray-core/discussions/2848#discussioncomment-8008806)
```
      {
        "type": "field",
        "port": "0-65535",
        "outboundTag": "warp"
      }
```

**Comment by pulsarice** [comment, marked as answer]
```
{
  "routing": {
    "domainStrategy": "IPOnDemand",
    "rules": [
      {
        "type": "field",
        "ip": [
          "x.x.x.x",
          "y.y.y.y"
        ],
        "outboundTag": "direct"
      },
      {
        "type": "field",
        "domain": [
          "domain:iptv1.com",
          "domain:iptv2.com"
        ],
        "outboundTag": "direct"
      },
      {
        "type": "field",
        "port": "0-65535",
        "outboundTag": "warp"
      }
    ]
  }
}
```
replace x.x.x.x and y.y.y.y with your IPTV IP addresses
replace iptv1.com and iptv2.com with your IPTV domain name.
"domainStrategy": "IPOnDemand" is important to make IP rules to take effect.
Your freedom outbound should have a "direct" tag:
```
{
  "tag": "direct",
  "protocol": "freedom"
}
```
